hardware_interface/include/hardware_interface/actuator.hpp

```diff
@@ -0,0 +1,87 @@
+// Copyright 2020 ROS2-Control Development Team
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#ifndef HARDWARE_INTERFACE__ACTUATOR_HPP_
+#define HARDWARE_INTERFACE__ACTUATOR_HPP_
+
+#include <algorithm>
+#include <memory>
+#include <string>
+#include <utility>
+
+#include "hardware_interface/component_info.hpp"
+#include "hardware_interface/component_interfaces/actuator_interface.hpp"
+#include "hardware_interface/types/hardware_interface_return_values.hpp"
+#include "hardware_interface/visibility_control.h"
+
+namespace hardware_interface
+{
+
+class Actuator final
```

Comment:
This looks good. When using this way of implementation is there a possibility to have shared functionality between actuators? And where should be this implemented:
examples:
* setting default members (name, n_dof, joint_names, etc.)
* managing of resource claims

Reply:
so the `final` class `Actuator` can do all sorts of things generic for all actuators. It also doesn't have to have the same API has the `ActuatorInterface`. So generic functionality could land in here.
I am not sure what you mean with "default members" though.

As for claiming resources, my initial thinking was that we could incorporate that in the configure step.

Reply:
„Default membership is not good description... sorry

I meant member which all actuator have since they are defined in urdf. Those are, ‚name‘, ‚joint‘! etc.


regarding configure, Init and other functions I will create state diagram tomorrow so we can really see what we need and what is done where. Just to be sure we are on the same page on this

Reply:
is that not what `ComponentInfo` is meant for?

Reply:
Ach ok, so you would not „parse“ values to the object itself, but add member of type ˋComponentInfoˋ.

This is OK for me!